src/node.cc

```diff
@@ -4978,6 +4990,324 @@ int Start(int argc, char** argv) {
+      LoadEnvironment(env);
+      env->async_hooks()->pop_async_id(1);
+    }
+
+    env->set_trace_sync_io(trace_sync_io);
+}
+
+v8::Local<v8::Value> Run(const std::string& path) {
+  // Read entire file into string. There is most certainly a better way ;)
+  // https://stackoverflow.com/a/2602258/2560557
+  std::ifstream t(path);
+  std::stringstream buffer;
+  buffer << t.rdbuf();
+
+  return Evaluate(buffer.str().c_str());
+}
+
+v8::Local<v8::Value> Evaluate(const std::string& java_script_code) {
+  EscapableHandleScope scope(env->isolate());
+  TryCatch try_catch(env->isolate());
+
+  // try_catch must be nonverbose to disable FatalException() handler,
+  // we will handle exceptions ourself.
+  try_catch.SetVerbose(false);
+
+  //ScriptOrigin origin(filename);
+  MaybeLocal<v8::Script> script = v8::Script::Compile(env->context(), v8::String::NewFromUtf8(isolate, java_script_code.c_str())/*, origin*/);
+  if (script.IsEmpty()) {
+    ReportException(env, try_catch);
+    exit(3);
+  }
+
+  Local<Value> result = script.ToLocalChecked()->Run();
+  if (result.IsEmpty()) {
+    ReportException(env, try_catch);
+    exit(4);
+  }
+
+  return scope.Escape(result);
+}
+
+void RunEventLoop(const RunUserLoop& callback){
+  bool more = false;
+  do {
+    more = ProcessEvents();
+    callback();
+  } while (more);
+}
+
+v8::Local<v8::Object> GetRootObject() {
+  return context->Global();
+}
+
+
+v8::Local<v8::Value> Call(v8::Local<v8::Object> receiver, v8::Local<v8::Function> function, const std::vector<v8::Local<v8::Value>> & args = {}) {
+    return function->Call(receiver, args.size(), const_cast<v8::Local<v8::Value>*>(&args[0]));
+}
+
+v8::Local<v8::Value> Call(v8::Local<v8::Object> receiver, v8::Local<v8::Function> function, std::initializer_list<v8::Local<v8::Value>> args) {
+    return Call(receiver, function, std::vector<v8::Local<v8::Value>>(args));
+}
+
+// TODO: Error handling: Node.js has exceptions disabled.
+v8::Local<v8::Value> Call(v8::Local<v8::Object> object, const std::string& function_name, const std::vector<v8::Local<v8::Value>>& args) {
+  Local<v8::String> v8_function_name = v8::String::NewFromUtf8(isolate, function_name.c_str());
+
+  Local<v8::Value> value = object->Get(v8_function_name);
+  if (!value->IsFunction()) {
+    //throw new Exception(":((");
+  }
+
+  return Call(object, v8::Local<v8::Function>::Cast(value), args);
+}
+
+v8::Local<v8::Value> Call(v8::Local<v8::Object> object, const std::string & function_name, std::initializer_list<v8::Local<v8::Value>> args) {
+    return Call(object, function_name, std::vector<v8::Local<v8::Value>>(args));
+}
+
+// TODO: Node.js has exceptions disabled.
+v8::Local<v8::Object> IncludeModule(const std::string& module_name) {
+  std::vector<v8::Local<v8::Value>> args = {v8::String::NewFromUtf8(isolate, module_name.c_str())};
+
+  auto module = Call(GetRootObject(), "require", args);
+  if (!module->IsObject()) {
+    // TODO: Can modules not be objects?
```

Comment:
This could be the case when the require fails, e.g. references a non-existing file/module. But if it is successful, modules are always objects.

Reply:
module.require() can also return Primitives